packages/superset-ui-core/src/query/extractExtras.ts

```diff
@@ -20,6 +21,8 @@ export default function extractExtras(formData: QueryFormData): Partial<QueryObj
     if (filter.col in reservedColumnsToQueryField) {
       const queryField = reservedColumnsToQueryField[filter.col];
       partialQueryObject[queryField] = filter.val;
+      // @ts-ignore
+      partialQueryObject.applied_time_extras[filter.col] = filter.val;
```

Comment:
Instead of `ts-ignore`, can use type assertion somewhere to make the typings compatible?

```typescript
      partialQueryObject.applied_time_extras[filter.col as QueryObjectAppliedTimeExtraKey] = filter.val;
```